server/src/main/java/io/crate/execution/dml/IndexDocumentBuilder.java

```diff
@@ -89,10 +89,12 @@ public Object getSyntheticValue(ColumnIdent columnIdent) {
 
     /**
      * Checks column constraints are all met for a given column and value
+     * @param noValue indicates that value is not coming from a user but a generated one.
+     * If value is generated, we can skip CheckGeneratedValue verification in order not to compare value with itself.
      */
-    public void checkColumnConstraint(ColumnIdent columnIdent, Object value) {
+    public void checkColumnConstraint(ColumnIdent columnIdent, Object value, boolean noValue) {
         Indexer.ColumnConstraint constraint = constraints.get(columnIdent);
-        if (constraint != null) {
+        if (constraint != null && (noValue && constraint instanceof Indexer.CheckGeneratedValue) == false) {
```

Comment:
Afaik the constraint could be a `MultiCheck`.

I'm generally not a fan of this approach. I'd rather fix the root cause.

Reply:
This is basically micro-optimization and just happens to be a bug fix.

Making `GeoShapeType.compareTo ` correct can fix it but then for inserting big generated Polygons with hundreds of points we would compare generated polygon with itself(!) per row. 

I agree that the root cause must be fixed but still think this is useful in general. 
Generated columns without provided value is a very common use case.

Reply:
> And we definitely should fix the root cause first, otherwise we're just masking bugs

To elaborate on this: if the user provides the values which match the generated value they'd still fail if we don't fix the root cause